tsconfig.paths.json

```diff
@@ -0,0 +1,625 @@
+/* This file is automatically generated by scripts/generators/tsconfig.js */
+{
+  "compilerOptions": {
+    "paths": {
```

Comment:
Interesting, i'm not sure why you are using ts paths here and not npm workspaces with exports
also using lib like wireit might help with eslint issue

Reply:
I... don't know :sweat_smile: It's what we were already doing before this PR. I guess it might be because we don't mark type-only dependencies as dependencies of packages, so the node_modules/workspaces structure is not complete for what TS needs.

Reply:
> so the node_modules/workspaces structure is not complete for what TS needs.
I didn't go this point, I think keeping what you have now but with using ts project refs and drop the ts-paths file makes more sense I guess...

Reply:
Avoiding paths would be great if you can; doing so effectively lies to the compiler about module layouts. Sometimes that's okay (especially for bundled projects), but I'm sorta surprised that this wouldn't "just work" if this repo is already a monorepo with package linking.

Reply:
If I remove paths TS complains that it cannot find the type definitions inside `node_modules` (probably because we store them in a separate directory) -- I'll keep using `paths` because TS is able to automatically map from the source files to their corresponding `.d.ts`.